ansible/roles/atmo-user-deployment-scripts/tasks/main.yml

```diff
@@ -15,8 +15,8 @@
 
 
 - name: deploy_script_2a - copy scripts to available scripts directory
-  copy:
-    content: "{{ item.text }}"
+  template:
+    src: "script.j2"
```

Comment:
Please figure out why the content isn't represented appropriately. My hunch is that `item.text` is a list of lines, if that is the case `{{ item.text | join('\n') }}` should work. But it should not be a list, did this issue crop up because of ansible 2.5?

Reply:
`item.text` was a str and would be copied into the file as a single line with literal `\n` characters in it:
```
#!/bin/bash\necho "wow" > /etc/wow.txt\necho "wow2" >> /etc/wow.txt
```

I made it into a list of lines using `splitlines()` in order to make this template approach work. However, your idea of using `{{ item.text | join('\n') }}` works and is better than using the template (I just tested it).

I did some tests using ansible 2.3 and 2.5 with a simple test playbook, and both copied content as expected so it is not a 2.5 issue. So it must be something about how the string is represented.

I believe it has something to do with [this method](https://github.com/cyverse/atmosphere/blob/37601a3397ee1f3bcadfe964ab7483abec8e82ea/core/models/boot_script.py#L74-L86) in `boot_script.py` so I will dig deeper there.

Reply:
In ansible, the `item.text` variable has `\\n` for newlines instead of `\n` which is why they are not evaluated. Just need to figure out why this happens so I need to learn more about the difference between str, unicode, and ascii

Reply:
Check if the boot script model is storing the literal newline character, i'm guessing it is.That code shouldn't have an effect newlines are in the ascii range.

Reply:
So it looks like the type of the script text is `str` in atmosphere, and it prints out in the logs just as expected. In the `execute_playbooks()` method, after creating the list of string args (including extra_vars), the script text has `\\n` instead of `\n`, as if using the `repr()` method. This was due to the way I created the list of strings: `--extra_vars=%s" % extra_vars`